src/prefect/utilities/logging.py

```diff
@@ -15,7 +15,19 @@
 import pendulum
 
 import prefect
-from prefect.configuration import config
+from prefect.utilities.context import context
+
+
+class PrefectLogger(logging.Logger):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.config = context.config.to_dict()
+
+    def __reduce__(self):
```

Comment:
Can you put a docstring on this? This feels like a very deep detail

Reply:
I'll be removing this entirely as worrying about pickling is no longer necessary with our new design.

Reply:
Removed.